### ISSUE TITLE
Tool testing improvements

### ISSUE THREAD

**Issue by mvdbeek** [galaxyproject/ephemeris#117]
These are a couple of things I found while implementing tool tests:

- tests only run of the test data is available from where the tests are triggered (this should be fixed with a galaxy-lib update containing https://github.com/galaxyproject/galaxy/pull/7160)
- tests seem to require an admin API key, that shouldn't be necessary with https://github.com/galaxyproject/galaxy/pull/7160
- We create a new history for every test, even if this test is for the same tool
- We just raise JobOutputExceptions, these need to be displayed properly
- We should have some reporting, ideally re-using some things from planemo

**Comment by selten** [comment]
@mvdbeek would it also be an idea to add the ability to remove the histories after all tests are finished (or at least as a command line option)?

**Comment by mvdbeek** [comment]
Yep, that would be a good option. I've added this along with some other things we've recently come across

**Comment by selten** [comment]
`shed-tools test` should have an option to exit with exit code 1 in case of failed tests. For me it seems to exit with exit code 0 even if it has failed tests.
